lib/jekyll/utils/internet.rb

```diff
@@ -0,0 +1,35 @@
+module Jekyll
+  module Utils
+    module Internet
+
+      # Public: Determine whether the present device has a connection to
+      # the Internet. This allows plugin writers which require the outside
+      # world to have a neat fallback mechanism for offline building.
+      #
+      # Example:
+      #   if Internet.connected?
+      #     Typhoeus.get("https://pages.github.com/versions.json")
+      #   else
+      #     Jekyll.logger.warn "Warning:", "Version check has been disabled."
+      #     Jekyll.logger.warn "", "Connect to the Internet to enable it."
+      #     nil
+      #   end
+      #
+      # Returns true if a DNS call can successfully be made, or false if not.
+      module_function
+      def connected?
+        !dns("google.com").nil?
```

Comment:
If you can find it, I will gladly update it! This is usually very fast and Google doesn't often botch their DNS. Seemed like the site I could think of with the best uptime!